yarn-project/pxe/src/note_processor/note_processor.ts

```diff
@@ -130,43 +153,48 @@ export class NoteProcessor {
         for (const functionLogs of txFunctionLogs) {
           for (const log of functionLogs.logs) {
             this.stats.seen++;
-            // @todo Issue(#6410) We should also try decrypting as outgoing if this fails.
-            const taggedNote = TaggedNote.decryptAsIncoming(log.data, secretKey);
-            if (taggedNote?.notePayload) {
-              const { notePayload: payload } = taggedNote;
-              // We have successfully decrypted the data.
+            const incomingTaggedNote = TaggedNote.decryptAsIncoming(log.data, ivskM)!;
+            const outgoingTaggedNote = TaggedNote.decryptAsOutgoing(log.data, ovskM)!;
+
+            if (incomingTaggedNote || outgoingTaggedNote) {
+              if (
+                incomingTaggedNote &&
+                outgoingTaggedNote &&
+                !incomingTaggedNote.notePayload.equals(outgoingTaggedNote.notePayload)
+              ) {
+                throw new Error('Incoming and outgoing note payloads do not match.');
+              }
+
+              const payload = incomingTaggedNote?.notePayload || outgoingTaggedNote?.notePayload;
+
               const txHash = block.body.txEffects[indexOfTxInABlock].txHash;
-              try {
-                const noteDao = await produceNoteDao(
-                  this.simulator,
-                  this.masterIncomingViewingPublicKey,
-                  payload,
-                  txHash,
-                  newNoteHashes,
-                  dataStartIndexForTx,
-                  excludedIndices,
-                );
-                noteDaos.push(noteDao);
+              const { incomingNote, outgoingNote, incomingDeferredNote } = await produceNoteDaos(
+                this.simulator,
+                incomingTaggedNote ? this.ivpkM : undefined,
+                outgoingTaggedNote ? this.ovpkM : undefined,
+                payload,
+                txHash,
+                newNoteHashes,
+                dataStartIndexForTx,
+                excludedIndices,
+                this.log,
+              );
+
+              if (incomingNote) {
+                incomingNotes.push(incomingNote);
+                this.stats.decrypted++;
+              }
+              if (outgoingNote) {
+                outgoingNotes.push(outgoingNote);
                 this.stats.decrypted++;
```

Comment:
Should there be different stats for decrypted incoming and outgoing?

Reply:
Addressed in https://github.com/AztecProtocol/aztec-packages/pull/6766/commits/daa6177de38bfcd7692a2e622ead66750b7f9ca9